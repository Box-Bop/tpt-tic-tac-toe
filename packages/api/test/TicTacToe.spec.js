const TicTacToe = require('../src/TicTacToe');

describe('TicTacToe', () => {

    it('2,3 x', () => {
        const t = new TicTacToe();
        t.makeTurn(0,2,'x');
        expect(t.getFields()[0][2]).toBe('x');
        expect(t.getFields()).toMatchSnapshot();
    });

    it('same turn two times', () => {
        const t = new TicTacToe();
        t.makeTurn(0,2,'x');
        expect(() => {
            t.makeTurn(0,2,'x');
        }).toThrow('spot already used');
    });

    it('not a valid coordinate', () => {
        const t = new TicTacToe();
        expect(() => {
            t.makeTurn(0,5,'x');
        }).toThrow('not a valid coordinate');
    });
    

  });
  