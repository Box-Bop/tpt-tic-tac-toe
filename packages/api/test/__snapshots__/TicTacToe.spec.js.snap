// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicTacToe 2,3 X 1`] = `
Array [
  Array [
    null,
    null,
    "X",
  ],
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
]
`;

exports[`TicTacToe victory 1 1`] = `
Array [
  Array [
    "X",
    "X",
    "X",
  ],
  Array [
    "O",
    "O",
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
]
`;

exports[`TicTacToe victory 2 1`] = `
Array [
  Array [
    "X",
    "O",
    null,
  ],
  Array [
    "X",
    "O",
    null,
  ],
  Array [
    "X",
    null,
    null,
  ],
]
`;
